src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const group: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const groupKey = `${item[key]}`;
+
+    if (group[groupKey] === undefined) {
```

Comment:
if the value is equal to `undefined`, it will be a runtime error.
Use another way to check, if a property exists in an object